### PR TITLE
Add New Relic Infrastructure API support

### PR DESCRIPTION
**Pull request by paul91**
As requested by @smithclay in https://github.com/terraform-providers/terraform-provider-newrelic/pull/30

Let me know if this is what you had in mind. This adds New Relic Infrastructure API support.

### CODE REVIEW

**Review by smithclay — COMMENTED** [review]
First review for the project, but this looks good to me.

@paultyng—would appreciate a quick spot check too since it's my first PR review in this repo :)

**Comment by petewilcock** [comment]
@paultyng Gentle ping here, just as I'm very keen to see these changes downstream in the NewRelic Terraform provider :)

File: api/client.go [hunk 1]
```diff
@@ -14,6 +14,20 @@ type Client struct {
 	RestyClient *resty.Client
 }
 
+// InfraClient represents the client state for the Infrastructure API
+type InfraClient struct {
```
Comment (paultyng) [inline api/client.go:18]: I'm not as familiar with this part of the New Relic API, so Infra is an entirely separate API (including a separate endpoint) I take it?

File: api/alert_infra_conditions.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
+package api
+
+import (
+	"fmt"
+	"net/url"
+	"strconv"
+)
+
+func (i *InfraClient) queryAlertInfraConditions(policyID int) ([]AlertInfraCondition, error) {
```
Comment (paultyng) [inline api/alert_infra_conditions.go]: `i` is probably a bad name for this due to its more idiomatic usage, looping, etc, maybe just `c` or `cli` or something that evokes client?

**Review by paultyng — CHANGES_REQUESTED** [review]
One minor change (rename `i`), but otherwise looks good to me. I would like there to be more assertions in the test coverage if possible, but not going to hold it up for that.

File: api/client.go (hunk 1 above)
Comment (smithclay) [reply api/client.go:18]: That's correct—separate endpoint.

**Comment by paul91** [comment]
Thanks for the feedback, I will address this later today.

**Comment by paul91** [comment]
@paultyng updated the InfraClient variable name to be more consistent with the rest of the project.

**Review by paultyng — APPROVED** [review]

**Comment by paultyng** [comment]
Released as v1.6.0

**Pull request merged**